### PR TITLE
Fixing issue #1072.

### PR DESCRIPTION
**Pull request by ntran18**
Due to the change in #1065, it breaks the import function of SIF files and GraphML files.

I added networkMode for the .sif file to determine "grn" or "protein-protein-physical-interaction." How to determine the network mode is mentioned in #1066. For the graphml files, the grnState.mode will always be "grn". 

Pull Request Checklist

- [x]  Travis C.I. build passes
- [x]  All five Demos look as expected when loaded
- [x]  Reload works as expected
- [x]  Loading file with Error brings up error modal, and it looks as expected
- [x]  Loading file with Warning brings up warnings modal
- [x]  Loading a network from the database looks as expected when loaded
- [x]  Import works for both SIF and GraphML
- [x]  Graph can be exported to SIF, GraphML and Excel
- [x]  Graph can be exported to PNG, SVG and PDF
- [x]  Expression data and Additional Sheets can be exported to Excel
- [x]  Print works as expected
- [x]  Restrict graph to viewport works as expected (check/uncheck)
- [x]  Viewport Size Changing works as expected (small/medium/large/fit)
- [x]  Toggle between Grid Layout and Force Graph Layout works as expected
- [x]  Force Graph Parameter Sliders change the number above them and have an effect on the graph
- [x]  Locking/Unlocking/Resetting/Undo Resetting the force graph parameters works as expected
- [x]  Enabling and disabling node coloring works as expected
- [x]  Node coloring options work as expected (selection top/bottom dataset, averaging values, changing max value)
- [x]  Weighted graph loaded with the "Enable Edge Coloring" option checked appears as expected
- [x]  Hide/Show Weights works as expected (always/never/upon mouseover)
- [x]  Edge Weight Normalization Factor can be changed (set/reset)
- [x]  Gray Edge Threshold can be changed, slider changes the number and has an effect on the graph
- [x]  Checking Show Gray Edges as Dashed works as expected (check/uncheck)
- [x]  D-pad left/right/top/bottom/center works as expected
- [x]  Zoom slider changes number and has effect on graph (viewport and menu)
- [x]  Right click on a node opens gene page, page is populated with correct data

### CODE REVIEW

**Comment by ceciliazaragoza** [comment]
- [ ]  Loading file with Error brings up error modal, and it looks as expected
  There is no error generated and graph is not generated for file: `test-files/semantic-error-test-files/empty-network-xlsx-with-network-tab.xlsx`. Furthermore, when other demos or files are selected, nothing displays.

**Comment by ntran18** [comment]
For the comment that @ceciliazaragoza mentioned, it wasn't my change. The code existed before, and we didn't catch that. I would suggest to create a new issue for this

**Comment by ceciliazaragoza** [comment]
Since we detected an issue here with the files, @akaiap and I can work on testing all the test-files on the current deployed beta to ensure that there are no other pre-existing bugs that need to be addressed.

File: server/controllers/sif-constants.js [hunk 1]
```diff
@@ -13,10 +13,10 @@ module.exports = {
...
             errorCode: "SIF_UNWEIGHTED_RELATIONSHIP_TYPE_ERROR",
             possibleCause: "The SIF importer detects an unweighted network with an unsupported relationship type.",
-            suggestedFix: "SIF files accepted by GRNsight must use 'pd' as the text string for the relationship" +
-                " type in unweighted networks. Please review the SIF input documentation. Additionally, this" +
-                " error may be have been caused by missing data in your file, which caused the importer to" +
-                " incorrectly interpret a source or target as the relationship."
+            suggestedFix: "SIF files accepted by GRNsight must use 'pd' or 'pp' as the text string for the" +
+                " relationship type in unweighted networks. Please review the SIF input documentation." +
+                " Additionally, this error may be have been caused by missing data in your file, which caused " +
+                "the importer to incorrectly interpret a source or target as the relationship."
```
Comment (dondi) [inline server/controllers/sif-constants.js:19]: Doesn’t have to change here to retain consistency, but for building new strings I have found that using the `join` array method feels structurally cleaner

**Review by dondi — APPROVED** [review]
LGTM

**Pull request merged**